### ISSUE TITLE
Add train script

### ISSUE THREAD

**Issue by Dayan-Zhanchi** [Lunkers/DD2412-Project#11]
This issue is more for implementing a general training script that can later be used for the 2 experiments, additive vs affine and Glow vs RealNVP.

Some sporadic observations regarding this:

- There's a preprocessing they do that converts the data into 5-bit or 8-bit, not sure. And I'm not even sure if this is used in the regular training
- Investigate how many epoches for the training -> depends on which experiment. Additive vs affine = 1800 epoches. RealNVP = no idea how many?
- There is no specific scheduling, learning rate or weight decay mentioned in the report, but the original code might have some. Although not sure if we should use it anyways. For instance weight decay is disbled by default.
- Need to add save and load logic for the model and checkpoints
- Tensorboard can be good to add aswell to keep track of training progression
- Figure out the batch size used => 512.
- Figure out hyperparameters for training -> both experiments should have basically the same
- Record how long the training takes, dunno if tensorboard can do this or we have to keep track with classic `tcurr = time.time()` method
- Figure out how to calculate loss function. In their model they have a seemingly more involved process for calculating the loss function than the simple one presented in the paper?

The original call to the train script is this:
`mpiexec -n 8 python train.py --problem cifar10 --image_size 32 --n_level 3 --depth 32 --flow_permutation 2 --flow_coupling 1 --seed 0 --learntop --lr 0.001 --n_bits_x 8`

So some hyperparameters are already given here that we can use.

**Comment by Lunkers** [comment]
<img width="764" alt="Skärmavbild 2020-10-08 kl  12 29 24" src="https://user-images.githubusercontent.com/24674133/95447382-e7c72d80-0961-11eb-9551-cc83f12a71e0.png">

**Comment by Lunkers** [comment]
This is the only thing we know about training optimization
